disco_aws_automation/disco_aws.py

```diff
@@ -308,21 +306,93 @@ def update_elb(self, hostclass, update_autoscaling=True, testing=False):
                 elb_subnets = self.get_subnets(elb_meta_network, hostclass)
                 subnet_ids = [subnet['SubnetId'] for subnet in elb_subnets]
 
-            elb_port = self.hostclass_option_default(hostclass, "elb_port", 80)
-            elb_protocol = self.hostclass_option_default(hostclass, "elb_protocol", None) or \
-                self._default_protocol_for_port(elb_port)
-            instance_port = int(self.hostclass_option_default(hostclass, "elb_instance_port", 80))
-            instance_protocol = self.hostclass_option_default(hostclass, "elb_instance_protocol", None) or \
-                self._default_protocol_for_port(instance_port)
+            def default_protocol_for_port(port):
+                """
+                Return the default protocl for the port
```

Comment:
One of the reasons I like closures is that I don't have to prefix function names with _ (I really wish Python had proper visibility control, or at least a keyword that signifies what _ does), but in this case I suppose that's the lesser evil.